src/extension/serializer.ts

```diff
@@ -801,6 +826,32 @@ export class GrpcSerializer extends SerializerBase {
     return notebook
   }
 
+  private static marshallFrontmatter(
+    metadata: { ['runme.dev/frontmatter']: string },
+    kernel?: Kernel,
+  ): Frontmatter {
+    const yamlDocs = YAML.parseAllDocuments(metadata['runme.dev/frontmatter'])
```

Comment:
Good question, @pastuxso! Initially, I started parsing the metadata with [gray-matter](https://www.npmjs.com/package/gray-matter), but since we were already using yaml, I thought it would be a good idea to stay consistent.

I think what matters most now is how our current design aligns with this approach, so using yaml should be fine for now, but I’ll leave the final decision to @sourishkrout.

Reply:
@peraltafederico do we have to serialize the frontmatter? `runme.parser.v1.Frontmatter` is well defined. If serialization is required on the backend side maybe that's where it should happen?

Reply:
@sourishkrout What we're using now as the Platform payload is the marshaled `Notebook` created at serialization time. We don't need to serialize the frontmatter, but we do need the frontmatter field to be populated in it (that way we can get the runme id in the Platform). So, I'm adding that extra marshal step for the frontmatter field, as we're doing now with the other fields. Makes sense? Let me know if I should follow a different pattern here!

Reply:
@sourishkrout Just to confirm, you mean that I just parse the raw frontmatter in the Platform APi instead of doing it when marshaling the Notebook, right?

Reply:
@sourishkrout  I just realized I'm marshaling the frontmatter to also get the session ID, see:
https://github.com/stateful/vscode-runme/blob/b56b841ffc5a54c47633ce1869ef74bd5a366da5/src/extension/serializer.ts#L829

I'm leaving this as is for now, but please let me know if there's a different approach I may be overlooking!

Reply:
Okay, @peraltafederico. We can leave it in place for now. It's in the protos but if we skip out on the Kernel reporter API for now, that's no help.